src/game/Protocol/WorldSocket.cpp

```diff
@@ -25,89 +26,179 @@
 #include "SharedDefines.h"
 #include "WorldSession.h"
 #include "WorldSocket.h"
-#include "WorldSocketMgr.h"
+
+#include <memory>
 #include "AddonHandler.h"
 #include "Opcodes.h"
-#include "MangosSocketImpl.h"
+#include "Auth/Sha1.h"
+#include "Database/SqlPreparedStatement.h"
+#include "Database/DatabaseEnv.h"
+#include "DBCStores.h"
 #include "ace/OS_NS_netdb.h"
+#include "WorldSocketMgr.h"
+
+#if defined( __GNUC__ )
+#pragma pack(1)
+#else
+#pragma pack(push,1)
+#endif
+struct ServerPktHeader
+{
+    uint16 size;
+    uint16 cmd;
+
+    const char* data() const {
+        return reinterpret_cast<const char*>(this);
+    }
 
-template class MangosSocket<WorldSession, WorldSocket, AuthCrypt>;
+    std::size_t headerSize() const {
+        return sizeof(ServerPktHeader);
+    }
+};
+#if defined( __GNUC__ )
+#pragma pack()
+#else
+#pragma pack(pop)
+#endif
 
-int WorldSocket::ProcessIncoming(WorldPacket* new_pct)
+WorldSocket::WorldSocket(IO::IoContext* ctx, IO::Networking::SocketDescriptor const& socketDescriptor)
+    : IO::Networking::AsyncSocket<WorldSocket>(ctx, socketDescriptor),
+            m_lastPingTime(std::chrono::system_clock::time_point::min()),
+            m_overSpeedPings(0),
+            m_Session(nullptr),
```

Comment:
Why is the S in m_Session capitalized, should be lower case.

Reply:
I tried to refactor the use of this variable, but there would be too many changes.
I will make a separate pr to `development` for this.